common-npm-packages/packaging-common/nuget/NuGetConfigHelper2.ts

```diff
@@ -146,6 +149,39 @@ export class NuGetConfigHelper2 {
         });
     }
 
+    private validateToolsVersion() {
+        let append = false;
+        let nugetVersion = tl.getInput('versionSpec', false);
```

Comment:
Is this the correct way to get the nuget version? I see the nuGetGetter import is unused

Reply:
I am not too sure tbh. 

1. When I look into NuGetGetter, none of the methods actually return me the currently in use Nuget version.  

2. I tried looking into tl.which('nuget') but that seems to be too many manual code to check.  After getting the nuget exe path,  there is not easy way to get the version. I have to trim off something or grep.
https://stackoverflow.com/questions/15964935/get-nuget-exe-version-from-command-line

3.  So I looked into nugettoolinstaller https://github.com/microsoft/azure-pipelines-tasks/blob/6ea095ec7e00d2c33b91c6a553ef3c587c509233/Tasks/XamariniOSV2/xamarinios.ts#L78 and see that in order to install a specific version, the env variable versionSpec will be set.  So assuming that the env variable versionSpec is **NOT** cleared between tasks' execution, I decided to get the version this way.

Reply:
Looks like other places in the code [call getFileVersionInfoAsync](https://github.com/microsoft/azure-pipelines-tasks/blob/92f8d738551328d50fefa04129fd0fce88456d92/Tasks/Common/packaging-common/pe-parser/index.ts#L12) with [a path to the executable](https://github.com/microsoft/azure-pipelines-tasks/blob/92f8d738551328d50fefa04129fd0fce88456d92/Tasks/NuGetCommandV2/nugetrestore.ts#L76)

Reply:
calling getInput will not do what you want.
* getInput gets an input to the currently executing task instance. Subsequent tasks in the pipeline will not contain that variable
* the versionSpec input in the tasks that take it may be an expression describing a _range_ of acceptable versions (a common one being something like "5.x")